fastjet/clustersequence.go

```diff
@@ -498,3 +499,44 @@ func (cs *ClusterSequence) runN3Dumb() error {
 	}
 	return err
 }
+
+// runNlnN runs the clustering using a Hierarchical Delaunay triangulation
+// and a min-heap to achieve O(N*ln N) behaviour.
+//
+// There are internally asserted assumptions about absence of points
+// with coincident eta-phi coordinates.
+func (cs *ClusterSequence) runNlnN() error {
+	panic(fmt.Errorf("fastjet: runNlnN not implemented"))
+}
+
+// addKtDistance adds the current kt distance for particle jeti to the heap
+// using information about the nearest neighbor in the eta-phi plane.
+// Work as follows:
+//
+// . if the kt is zero then it's nearest neighbour is taken to be the
+//   the beam jet and the distance is zero.
+//
+// . if cylinder distance to nearest neighbour > _Rparam then it is
```

Comment:
`_Rparam`... this looks like a C++ private member :)

Reply:
Oops :)

Reply:
Would that be r^2 in our case?

Reply:
yep.